pallets/aleph/src/lib.rs

```diff
@@ -284,9 +285,47 @@ pub mod pallet {
         ) -> DispatchResult {
             ensure_root(origin)?;
 
-            let azero_cap = azero_cap.unwrap_or_else(AzeroCap::<T>::get);
-            let horizon_millisecs =
-                horizon_millisecs.unwrap_or_else(ExponentialInflationHorizon::<T>::get);
+            let current_azero_cap = AzeroCap::<T>::get();
+            let current_horizon_millisecs = ExponentialInflationHorizon::<T>::get();
+            let total_issuance = T::TotalIssuanceProvider::get();
+
+            let azero_cap = azero_cap.unwrap_or(current_azero_cap);
+            let horizon_millisecs = horizon_millisecs.unwrap_or(current_horizon_millisecs);
+
+            let current_gap = current_azero_cap.saturating_sub(total_issuance);
+            let new_gap = match azero_cap.checked_sub(total_issuance) {
+                Some(new_gap) => new_gap,
+                None => {
+                    return Err(DispatchError::Other(
+                        "AZERO Cap cannot be lower than the current total issuance!",
+                    ))
+                }
+            };
+
+            if horizon_millisecs
```

Comment:
Mv  every if to function and name it so reader know immediately what's the check

Reply:
done